CMakeLists.txt

```diff
@@ -0,0 +1,342 @@
+cmake_minimum_required(VERSION 3.28)
+
+cmake_policy(SET CMP0048 NEW)
+cmake_policy(SET CMP0091 NEW)
+
+set(PKG_VERSION 1.0.31)
+set(LIB_VERSION 1.74.0)
+
+project(tkrzw
+  VERSION ${PKG_VERSION}
+  DESCRIPTION "a set of implementations of DBM"
+  HOMEPAGE_URL "https://dbmx.net/tkrzw/"
+  LANGUAGES C CXX
+)
+add_library(tkrzw)
+set_target_properties(tkrzw PROPERTIES VERSION ${LIB_VERSION} SOVERSION 1)
+
+include(CMakePackageConfigHelpers)
+include(CPack)
+include(GNUInstallDirs)
+include(CMakeDependentOption)
+
+option(WITH_TOOLS "Build cli frontends" OFF)
+cmake_dependent_option(WITH_TESTS "Generate a test target" OFF "WITH_TOOLS" OFF)
+
+
+find_package(Threads)
+target_link_libraries(tkrzw PUBLIC Threads::Threads)
+
+# See https://gitlab.kitware.com/cmake/cmake/-/issues/18837
+if(MSVC)
+	target_compile_options(tkrzw PUBLIC "/Zc:__cplusplus")
+endif()
+
+target_compile_definitions(tkrzw
+    PRIVATE
+        _FILE_OFFSET_BITS=64
+        _TKRZW_PKG_VERSION="${CMAKE_PROJECT_VERSION}"
+        _TKRZW_LIB_VERSION="${LIB_VERSION}"
+)
+
+target_compile_features(tkrzw PUBLIC cxx_std_17)
+
+target_include_directories(tkrzw
+    INTERFACE
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/>
+)
+set_target_properties(tkrzw
+    PROPERTIES
+        MSVC_RUNTIME_LIBRARY MultiThreaded$<$<CONFIG:Debug>:Debug>$<IF:$<STREQUAL:$<TARGET_PROPERTY:tkrzw,TYPE>,SHARED_LIBRARY>,DLL,>
```

Comment:
I don't have access to Windows machine now, but last time I checked for relative newer CMake, this is somewhat useless?

Reply:
No?

[CMP0091](https://cmake.org/cmake/help/v3.28/policy/CMP0091.html) page from v3.28 says 

> Use the [cmake_policy()](https://cmake.org/cmake/help/v3.28/command/cmake_policy.html#command:cmake_policy) command to set it to OLD or NEW explicitly.


The [cmake_minimum_required](https://cmake.org/cmake/help/v3.28/command/cmake_minimum_required.html#policy-settings) page says

> All policies known to the running version of CMake and introduced in the <min> (or <max>, if specified) version or earlier will be set to use NEW behavior.

I think this means `cmake_minimum_required(VERSION 3.28)` will set `NEW` for all the policies introduced in `VERSION <= 3.28`.